presto-main/src/main/java/com/facebook/presto/sql/planner/TranslationMap.java

```diff
@@ -245,24 +249,24 @@ public Expression rewriteIdentifier(Identifier node, Scope context, ExpressionTr
                     return coerceIfNecessary(node, symbol.toSymbolReference());
                 }
                 else {
-                    return rewriteExpressionWithResolvedName(node);
+                    return rewriteExpressionWithResolvedName(node, context);
                 }
             }
 
-            private Expression rewriteExpressionWithResolvedName(Expression node)
+            private Expression rewriteExpressionWithResolvedName(Expression node, Scope context)
             {
-                return getSymbol(rewriteBase, node)
+                return getSymbol(rewriteBase, context, node)
                         .map(symbol -> coerceIfNecessary(node, symbol.toSymbolReference()))
                         .orElse(coerceIfNecessary(node, node));
             }
 
             @Override
             public Expression rewriteDereferenceExpression(DereferenceExpression node, Scope context, ExpressionTreeRewriter<Scope> treeRewriter)
             {
-                Optional<ResolvedField> resolvedField = rewriteBase.getScope().tryResolveField(node);
```

Comment:
I guess you could just make sure that `analysis.getLambdaArgumentReference(node)==null` before trying to resolve using scope or this would be too simplistic?